packages/components/src/Heading/Heading.tsx

```diff
@@ -0,0 +1,81 @@
+import React, { ReactNode } from "react";
+import {
+  TypographyColor,
+  TypographySize,
+  styleFromColor,
+  styleFromSize,
+} from "../util/typography";
+import tw, { styled } from "twin.macro";
+
+type HeadingElement = "h1" | "h2" | "h3" | "h4" | "h5" | "h6";
+
+type Props = {
+  /**
+   * This prop can be used to specify which size heading should
+   * actually be rendered, in the case that you want to render an element
+   * as one heading but style it as if it were another. If both an `as` prop
+   * and a `size` prop are passed, the `as` will be used to determine the element
+   * and the `size` will be used to determine the styling.
+   */
+  as?: HeadingElement;
+  /**
+   * Specifies font weight as either bold or normal (defaults to bold).
+   * TODO: Currently we will allow toggling any size, but in the future
+   * we will add stricter enforcement -- e.g. enforcing a boldness
+   * for each size or checking for mutually exclusive props.
+   */
+  bold?: boolean;
+  /**
+   * The text content to present.
+   */
+  children: ReactNode;
+  /**
+   * The color of the heading element. If no color provided, defaults to a base color.
+   */
+  color?: TypographyColor;
+  /**
+   * The size of the html element. If no `as` prop is provided, then
+   * the component uses this value to determine which html tag to render
+   * (e.g. 'h1', 'h2', etc.)
+   */
+  size: TypographySize;
+};
+
+interface BaseHeadingProps {
+  bold: boolean;
+  color?: TypographyColor;
+  size: TypographySize;
+}
+
+const HeadingComponent = styled.span<BaseHeadingProps>(
```

Comment:
Just realized that even though this says `styled.span`, it'll get correctly overriden by the `as`/`size` props.

Musing out loud here: I wonder if it would make sense to split the styles for the different heading levels.

```
const H1 = styled.h1<BaseHeadingProps>`
  font-size: 32px // or whatever it is
`;

const H2 = styled.h2<BaseHeadingProps>(...);
const H3 = styled.h3<BaseHeadingProps>(...);
// etc
```

Moving away from dynamic styles, and towards static styles where possible has a couple benefits
- Easier to read and understand, in my opinion. Especially as more of them get added.
- There's less runtime cost. And if we can get to all static for a component, there's no runtime cost.

Like I said, just thinking out loud, and things to think and talk about. Nothing to do for now.

Reply:
thanks for the musing 👍 I thought about this previously, but I wasn't sure how much we could actually save since
- the element can still be changed by `as`
- there's still dynamic `color`/`bold` props

I hadn't considered readability though! and using static styles where possible is def a good general approach